refactor(ci): move to poetry

cpcloud opened the pull request.

pep8speaks commented:
Hello @cpcloud! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:









* In the file [`setup.py`](https://github.com/cpcloud/ibis/blob/e8f23ded7dde82f28aa68e1fc2bf4c289d9489c3/setup.py):

> [Line 131:80](https://github.com/cpcloud/ibis/blob/e8f23ded7dde82f28aa68e1fc2bf4c289d9489c3/setup.py#L131): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (2082 > 79 characters)



##### Comment last updated at 2021-09-20 12:54:01 UTC

cpcloud committed:
refactor(ci): move to poetry

kszucs committed:
refactor(ir): move ibis.sequence() to api.py (#3025)

* Move ibis.sequence() to api.py

* Clean up case builders

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix value order with set predicate

* Additional operations with ValueLust arguments

* Set ordering in impala and clickhouse tests

* Set ordering in impala and clickhouse tests

* Pre-commit fix

* Remove TODO notest

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

kszucs committed:
refactor(signature.py): remove the need of `Argument` constructor (#3047)

* Signature refactor

* Signature refactor

* Signature refactor

* Signature refactor

* Examples

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix visualization

* Flake8

* Restore ValueList rules

* Cleanup

* Fix impala udf creation

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Linting

* Address review comments

* Add test for argument reordering

* Finalize

* Restore docstring

* ABC

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

cpcloud committed:
refactor(rules.py): remove most operation constructors (#3046)

* refactor: remove most operation constructors

* refactor: reuse sort_key logic

* refactor: remove custom constructor for CrossJoin entirely

* refactor: bring back `prefixes` check for backcompat

* refactor: remove column_name_from

* fix: handle the empty schema case in _table_to_array

* refactor: move _flatten to util.py and add tests

* fix: check for ScalarExpr in named_literal_expression

* fix: allow only columns for group by keys

* refactor: move literal to map_to

* style: rename named_literal_expression to named_literal

* refactor: remove ir.TableExpr refs in rules

* refactor: deepcopy the default to optional to prevent mutation

* fix: make sure set test doesn't depend on ordering of expected output

* test: add test for copying default value
